internal/libhive/api.go

```diff
@@ -218,6 +218,13 @@ func (api *simAPI) startClient(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Get the network names, if any, for the container to be connected to at start
+	networks, err := api.checkClientNetworks(r, w)
+	if err != nil {
+		// We return here so we don't even create the container in bad requests
+		return
```

Comment:
You need to return something to the client here, otherwise it will get status code 200 and no response body. We usually also log something in this case. See line 195 for an example.

Reply:
Oh wait, I see the function `checkClientNetworks` already sets the response.

Reply:
I would prefer to handle / log the error here, i.e. don't pass the response to `checkClientNetworks`.

Reply:
Yes, I agree, it looks confusing having a simple return here. Will change it no problem.

Reply:
Should I take the opportunity to also change lines 216-219 to use the same pattern ?

Reply:
Why not!